### PR TITLE
MARP-825 Release 11.3

### PR DESCRIPTION
**Pull request by ntqdinh-axonivy**

### CODE REVIEW

File: ldap-connector-test/config/variables.yaml [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 # add an additional ‘variables.yaml’ file in a subdirectory in the ‘Config’ folder: 
 # '<project>/Config/_<environment>/variables.yaml
 #
 Variables:
   LdapConnector:
-    Url: "ldap://159.100.249.76:389"
+    Url: ldap://localhost:1389
+    Provider: com.sun.jndi.ldap.LdapCtxFactory
+    Referal: follow
+    Connection.Timeout: "1000"
```
Comment (phhung-axonivy) [inline ldap-connector-test/config/variables.yaml:13]: Split this variable to 2 lines

File: ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java [hunk 2]
```diff
@@ -56,15 +69,36 @@ static void setupConfig() throws IOException {
...
             .connectionTimeout(Ivy.var().get("LdapConnector.Connection.Timeout"))
             .provider(Ivy.var().get("LdapConnector.Provider"))
             .referral(Ivy.var().get("LdapConnector.Referral"))
             .toJndiConfig();
     queryExecutor = new LdapQueryExecutor(config);
     writer = new LdapWriter(config);
+    
+// Setup docker for testing
+    Network network = Network.newNetwork();
+    ldapContainer = new GenericContainer<>(LDAP_IMAGE).withNetwork(network).withNetworkAliases("octopus_ldap")
```
Comment (phhung-axonivy) [inline ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java:83]: Check warning and resolve it

**Review by phhung-axonivy — CHANGES_REQUESTED** [review]

**Review by phhung-axonivy — APPROVED** [review]
LGTM

**Comment by ntqdinh-axonivy** [comment]
Hi @ivy-lm @ivy-rew 
We have a request to release a new version of engine 11.3 for this connector. To fix failed tests, I use the latest open LDAP images from Bitami (powered by VMware), set up default db/schema & reword test cases a little bit instead of building a real instance of an Active Directory. 

I also disabled the option to skip failed tests for CI/CD workflow but not sure if it is appropriate or not.

File: ldap-connector-test/src_test/resources/docker/ldifs/data.ldif [hunk 3]
```diff
@@ -0,0 +1,36 @@
+version: 1
```
Comment (ivy-rew) [inline ldap-connector-test/src_test/resources/docker/ldifs/data.ldif:1]: these data.ldif and octopus.ldif files are now twice in the SCM ... we should only maintain one version.

File: ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java [hunk 4]
```diff
@@ -56,15 +70,36 @@ static void setupConfig() throws IOException {
...
             .connectionTimeout(Ivy.var().get("LdapConnector.Connection.Timeout"))
             .provider(Ivy.var().get("LdapConnector.Provider"))
             .referral(Ivy.var().get("LdapConnector.Referral"))
             .toJndiConfig();
     queryExecutor = new LdapQueryExecutor(config);
     writer = new LdapWriter(config);
+    
+    // Setup docker for testing
+    Network network = Network.newNetwork();
+    ldapContainer = new GenericContainer<>(LDAP_IMAGE).withNetwork(network).withNetworkAliases("octopus_ldap")
```
Comment (ivy-rew) [inline ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java:84]: any feedback to team Octopus regarding test-containers? @alexsuter

**Comment by ivy-rew** [comment]
> Hi @ivy-lm @ivy-rew We have a request to release a new version of engine 11.3 for this connector. To fix failed tests, I use the latest open LDAP images from Bitami (powered by VMware), set up default db/schema & reword test cases a little bit instead of building a real instance of an Active Directory.
> 
> I also disabled the option to skip failed tests for CI/CD workflow but not sure if it is appropriate or not.

fyi @ivy-lmu

File: ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java (hunk 4 above)
Comment (alexsuter) [reply ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java:84]: using test-containers is a good idea!

**Review by ivy-rew — APPROVED** [review]
very nice isolation of the complex infra in docker :1st_place_medal:

File: ldap-connector-test/src_test/resources/docker/ldifs/data.ldif (hunk 3 above)
Comment (ntqdinh-axonivy) [reply ldap-connector-test/src_test/resources/docker/ldifs/data.ldif:1]: The folder "ldifs" is used to set up default db for testing (like person_query or group_query test case) while octopus.ldif is additional db schema of this instance (like group and user class). 

I tried to combine 2 of them 1 one file but there is no hint to that.

How if I rename the schema file to make it more meaningful & not make any confuse for us?

Comment (ivy-rew) [reply ldap-connector-test/src_test/resources/docker/ldifs/data.ldif:1]: it's totally ok to have these 2 files. 
But it seems as if you have the same files twice under version control:
- src_test/resources/docker/ldifs 
- docker/ldifs
IMHO one of these directories should be removed.

Comment (ntqdinh-axonivy) [reply ldap-connector-test/src_test/resources/docker/ldifs/data.ldif:1]: Oops, my bad. I will check & remove one of these folders soon.

**Pull request merged**